src/Interpreters/Context.cpp

```diff
@@ -740,12 +740,16 @@ struct ContextSharedPart : boost::noncopyable
 
     void initializeTraceCollector(std::shared_ptr<TraceLog> trace_log)
     {
-        if (!trace_log)
-            return;
+        chassert(trace_collector.has_value());
```

Comment:
better to throw LOGIAL_ERROR here because trace collector is not used in non release builds